spec/integration/strong_resource_spec.rb

```diff
@@ -132,21 +132,53 @@
           params[:person][:company_attributes] = {
             title: 'walmart',
             state_attributes: {
-              acronym: 'ny'
-            }
+            acronym: 'ny'
+          }
           }
         end
 
         it 'allows nesting relations' do
           expect(controller.strong_resource.to_h)
-            .to eq('name' => 'John', 'company_attributes' => {
-              'title' => 'walmart',
-              'state_attributes' => {
-                'acronym' => 'ny'
-              }
-            })
+          .to eq('name' => 'John', 'company_attributes' => {
+            'title' => 'walmart',
+            'state_attributes' => {
+            'acronym' => 'ny'
+          }
+          })
         end
       end
     end
   end
 end
+
+describe "strong_resources" do
+  context "with multiple definitions" do
+    let(:controller) do
+      controller = CompaniesController.new
+      controller.params = ActionController::Parameters.new(params)
+      allow(controller).to receive(:action_name) { :create }
+      controller
+    end
+
+    let(:params) do
+      { parent_company: { title: 'John' }, data: { type: 'parent_company' } }
```

Comment:
looks like the type is never changing...shouldnt we be testing 'company'within this test as well?